### ISSUE TITLE
database init via run.sh?

### ISSUE THREAD

**Issue by woebbi** [fake-name/xA-Scraper#11]
Hey just wondering if the db creation is triggered via the run.sh (provided I didn't screw up in the setup of my postgresql) or do I have to run a different .py?

**Comment by fake-name** [comment]
Hm, I rewrote a giant pile of everything recently, so the docs are probably not correct at this point. Some core features are currently kind of broken too (I haven't gotten around to fixing the add-artist bits, whoops?).

You are going to want to init the db via `python3 db_migrate.py db migrate` (I think, from memory).

**Comment by jimhsu** [comment]
Running db migrate produces: 

  File "/home/myblocks/.local/lib/python3.5/site-packages/alembic/runtime/environment.py", line 836, in run_migrations
    self.get_context().run_migrations(**kw)
  File "/home/myblocks/.local/lib/python3.5/site-packages/alembic/runtime/migration.py", line 321, in run_migrations
    for step in self._migrations_fn(heads, self):
  File "/home/myblocks/.local/lib/python3.5/site-packages/alembic/command.py", line 156, in retrieve_migrations
    revision_context.run_autogenerate(rev, context)
  File "/home/myblocks/.local/lib/python3.5/site-packages/alembic/autogenerate/api.py", line 415, in run_autogenerate
    self._run_environment(rev, migration_context, True)
  File "/home/myblocks/.local/lib/python3.5/site-packages/alembic/autogenerate/api.py", line 427, in _run_environment
    raise util.CommandError("Target database is not up to date.")
alembic.util.exc.CommandError: Target database is not up to date.

Running db upgrade after that produces:

...
  File "/home/myblocks/.local/lib/python3.5/site-packages/sqlalchemy/util/compat.py", line 186, in reraise
    raise value.with_traceback(tb)
  File "/home/myblocks/.local/lib/python3.5/site-packages/sqlalchemy/engine/base.py", line 1182, in _execute_context
    context)
  File "/home/myblocks/.local/lib/python3.5/site-packages/sqlalchemy/engine/default.py", line 470, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) relation "siteartistnames" does not exist
LINE 7:         FROM siteartistnames;
                     ^
 [SQL: '\n        SELECT\n             sitename,\n             artistname,\n             uploadeh,\n             last_fetched\n        FROM siteartistnames;\n        ']

**Comment by fake-name** [comment]
Yeah, the migration is broken because it's assuming the older database layout is present. It needs to check if older tables exist before migrating. 

TL;DR DERP DERP DERP DERP.

**Comment by fake-name** [comment]
Ok, bootstrapping from a clean DB now works.

Note that I still need to reimplement the mechanism for adding scrape targets. Since I tore out and rewrote most of the web-ui backing recently, I haven't reimplemented it yet.
